### ISSUE TITLE
macOS/Conda CI failure Mid April 2022

### ISSUE THREAD

**Issue by traversaro** [robotology/robotology-superbuild#1090]
The CMake configuration error is:
~~~
2022-04-13T02:44:37.6750160Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:37.6850600Z   HEADER_SETS property is read-only
2022-04-13T02:44:37.6950270Z 
2022-04-13T02:44:37.7050860Z Call Stack (most recent call first):
2022-04-13T02:44:37.7151820Z   src/yarpview/src/CMakeLists.txt:72 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:37.7251560Z 
2022-04-13T02:44:37.7351940Z 
2022-04-13T02:44:37.7452830Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:37.7553610Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:37.7653230Z 
2022-04-13T02:44:37.7754060Z Call Stack (most recent call first):
2022-04-13T02:44:37.7854490Z   src/yarpview/src/CMakeLists.txt:72 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:37.7954260Z 
2022-04-13T02:44:37.8054800Z 
2022-04-13T02:44:37.8160920Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:37.8262400Z   HEADER_SETS property is read-only
2022-04-13T02:44:37.8363240Z 
2022-04-13T02:44:37.8464040Z Call Stack (most recent call first):
2022-04-13T02:44:37.8565160Z   src/yarpscope/src/CMakeLists.txt:75 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:37.8665860Z 
2022-04-13T02:44:37.8766390Z 
2022-04-13T02:44:37.8867270Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:37.8968790Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:37.9069540Z 
2022-04-13T02:44:37.9181060Z Call Stack (most recent call first):
2022-04-13T02:44:37.9317520Z   src/yarpscope/src/CMakeLists.txt:75 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:37.9428180Z 
2022-04-13T02:44:37.9530400Z 
2022-04-13T02:44:37.9632330Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:37.9751690Z   HEADER_SETS property is read-only
2022-04-13T02:44:37.9858610Z 
2022-04-13T02:44:37.9963660Z Call Stack (most recent call first):
2022-04-13T02:44:38.0065160Z   src/yarpmanager/CMakeLists.txt:195 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.0166050Z 
2022-04-13T02:44:38.0311890Z 
2022-04-13T02:44:38.0725410Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.0830510Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:38.0932140Z 
2022-04-13T02:44:38.1033740Z Call Stack (most recent call first):
2022-04-13T02:44:38.1755430Z   src/yarpmanager/CMakeLists.txt:195 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.1857760Z 
2022-04-13T02:44:38.1961650Z 
2022-04-13T02:44:38.2064250Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.2165700Z   HEADER_SETS property is read-only
2022-04-13T02:44:38.2290790Z 
2022-04-13T02:44:38.2393450Z Call Stack (most recent call first):
2022-04-13T02:44:38.2495990Z   src/yarplogger/CMakeLists.txt:79 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.2596830Z 
2022-04-13T02:44:38.2716750Z 
2022-04-13T02:44:38.2818520Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.2927060Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:38.3031430Z 
2022-04-13T02:44:38.3132520Z Call Stack (most recent call first):
2022-04-13T02:44:38.3234040Z   src/yarplogger/CMakeLists.txt:79 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.3337170Z 
2022-04-13T02:44:38.3437770Z 
2022-04-13T02:44:38.3540030Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.3658160Z   HEADER_SETS property is read-only
2022-04-13T02:44:38.3785100Z 
2022-04-13T02:44:38.3900690Z Call Stack (most recent call first):
2022-04-13T02:44:38.4027370Z   src/yarpdataplayer/CMakeLists.txt:134 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.4128190Z 
2022-04-13T02:44:38.4229060Z 
2022-04-13T02:44:38.4330300Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.4432010Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:38.4533780Z 
2022-04-13T02:44:38.4643210Z Call Stack (most recent call first):
2022-04-13T02:44:38.4746270Z   src/yarpdataplayer/CMakeLists.txt:134 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.4860620Z 
2022-04-13T02:44:38.4960970Z 
2022-04-13T02:44:38.5061800Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.5205790Z   HEADER_SETS property is read-only
2022-04-13T02:44:38.5307380Z 
2022-04-13T02:44:38.5408450Z Call Stack (most recent call first):
2022-04-13T02:44:38.5561320Z   src/yarpmotorgui/CMakeLists.txt:106 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.5662120Z 
2022-04-13T02:44:38.5763860Z 
2022-04-13T02:44:38.5865830Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.5968070Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:38.6069320Z 
2022-04-13T02:44:38.6171000Z Call Stack (most recent call first):
2022-04-13T02:44:38.6299260Z   src/yarpmotorgui/CMakeLists.txt:106 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.6400070Z 
2022-04-13T02:44:38.6501110Z 
2022-04-13T02:44:38.6612470Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.6715300Z   HEADER_SETS property is read-only
2022-04-13T02:44:38.6816040Z 
2022-04-13T02:44:38.6917010Z Call Stack (most recent call first):
2022-04-13T02:44:38.7018970Z   src/yarpbatterygui/CMakeLists.txt:86 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.7119830Z 
2022-04-13T02:44:38.7220510Z 
2022-04-13T02:44:38.7321480Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.7422790Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:38.7523540Z 
2022-04-13T02:44:38.7624410Z Call Stack (most recent call first):
2022-04-13T02:44:38.7725490Z   src/yarpbatterygui/CMakeLists.txt:86 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.7826260Z 
2022-04-13T02:44:38.7926790Z 
2022-04-13T02:44:38.8027750Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.8072320Z   HEADER_SETS property is read-only
2022-04-13T02:44:38.8172570Z 
2022-04-13T02:44:38.8273360Z Call Stack (most recent call first):
2022-04-13T02:44:38.8373740Z   src/yarpmobilebasegui/CMakeLists.txt:91 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.8474430Z 
2022-04-13T02:44:38.8574760Z 
2022-04-13T02:44:38.8675700Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.8776270Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:38.8876510Z 
2022-04-13T02:44:38.8977300Z Call Stack (most recent call first):
2022-04-13T02:44:38.9077740Z   src/yarpmobilebasegui/CMakeLists.txt:91 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.9178390Z 
2022-04-13T02:44:38.9278880Z 
2022-04-13T02:44:38.9379920Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:38.9480720Z   HEADER_SETS property is read-only
2022-04-13T02:44:38.9580120Z 
2022-04-13T02:44:38.9680990Z Call Stack (most recent call first):
2022-04-13T02:44:38.9782050Z   src/yarplaserscannergui/CMakeLists.txt:72 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:38.9881960Z 
2022-04-13T02:44:38.9982550Z 
2022-04-13T02:44:39.0083360Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:39.0184030Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:39.0283700Z 
2022-04-13T02:44:39.0385070Z Call Stack (most recent call first):
2022-04-13T02:44:39.0485550Z   src/yarplaserscannergui/CMakeLists.txt:72 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:39.0585330Z 
2022-04-13T02:44:39.0685670Z 
2022-04-13T02:44:39.0786550Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:39.0887080Z   HEADER_SETS property is read-only
2022-04-13T02:44:39.0986940Z 
2022-04-13T02:44:39.1088070Z Call Stack (most recent call first):
2022-04-13T02:44:39.1188270Z   src/yarpviz/src/CMakeLists.txt:100 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:39.1288680Z 
2022-04-13T02:44:39.1388850Z 
2022-04-13T02:44:39.1489890Z CMake Error at cmake/YarpMacOSUtilities.cmake:101 (set_target_properties):
2022-04-13T02:44:39.1590380Z   INTERFACE_HEADER_SETS property is read-only
2022-04-13T02:44:39.1690050Z 
2022-04-13T02:44:39.1790720Z Call Stack (most recent call first):
2022-04-13T02:44:39.1891370Z   src/yarpviz/src/CMakeLists.txt:100 (yarp_macos_duplicate_and_add_bundle)
2022-04-13T02:44:39.1991570Z 
2022-04-13T02:44:39.2092020Z 
~~~

This is probably a macOS/CMake 3.23.1 regression as this CMake version landed in conda-forge yesterday: https://github.com/conda-forge/cmake-feedstock/pull/159 , while homebrew still has CMake 3.23.0 .

**Comment by traversaro** [comment]
Fix provided in https://github.com/robotology/yarp/pull/2822 .

**Comment by traversaro** [comment]
> Fix provided in [robotology/yarp#2822](https://github.com/robotology/yarp/pull/2822) .

While waiting for the fix, I provided the workaround in https://github.com/robotology/robotology-superbuild/pull/1092 that should fix the CI and prevent failures for macOS users .

**Comment by traversaro** [comment]
Fixed in https://github.com/robotology/robotology-superbuild/pull/1092, long term fix proposed in https://github.com/robotology/yarp/pull/2822 .
